libraries/standard/http/src/http_client.c

```diff
@@ -40,21 +139,149 @@ HTTPStatus_t HTTPClient_InitializeRequestHeaders( HTTPRequestHeaders_t * pReques
 /*-----------------------------------------------------------*/
 
 HTTPStatus_t HTTPClient_AddHeader( HTTPRequestHeaders_t * pRequestHeaders,
-                                   const char * pName,
-                                   size_t nameLen,
+                                   const char * pField,
+                                   size_t fieldLen,
                                    const char * pValue,
                                    size_t valueLen )
 {
-    return HTTP_NOT_SUPPORTED;
+    HTTPStatus_t returnStatus = HTTP_SUCCESS;
+
+    /* Check for NULL parameters. */
+    if( pRequestHeaders == NULL )
+    {
+        IotLogError( "Parameter check failed: pRequestHeaders interface is NULL." );
+        returnStatus = HTTP_INVALID_PARAMETER;
+    }
+    else if( pRequestHeaders->pBuffer == NULL )
+    {
+        IotLogError( "Parameter check failed: pRequestHeaders->pBuffer is NULL." );
+        returnStatus = HTTP_INVALID_PARAMETER;
+    }
+    else if( ( pField == NULL ) )
+    {
+        IotLogError( "Parameter check failed: pField is NULL." );
+        returnStatus = HTTP_INVALID_PARAMETER;
+    }
+    else if( ( pValue == NULL ) )
+    {
+        IotLogError( "Parameter check failed: pValue is NULL." );
+        returnStatus = HTTP_INVALID_PARAMETER;
+    }
+    else if( fieldLen == 0u )
+    {
+        IotLogError( "Parameter check failed: fieldLen must be greater than 0." );
+        returnStatus = HTTP_INVALID_PARAMETER;
+    }
+    else if( valueLen == 0u )
+    {
+        IotLogError( "Parameter check failed: valueLen must be greater than 0." );
+        returnStatus = HTTP_INVALID_PARAMETER;
+    }
+    else
+    {
+        /* Empty else MISRA 15.7 */
+    }
+
+    if( returnStatus == HTTP_SUCCESS )
+    {
+        returnStatus = _addHeader( pRequestHeaders,
+                                   pField, fieldLen, pValue, valueLen );
+    }
+
+    return returnStatus;
 }
 
 /*-----------------------------------------------------------*/
 
 HTTPStatus_t HTTPClient_AddRangeHeader( HTTPRequestHeaders_t * pRequestHeaders,
-                                        int32_t rangeStart,
+                                        int32_t rangeStartOrlastNbytes,
                                         int32_t rangeEnd )
 {
-    return HTTP_NOT_SUPPORTED;
+    HTTPStatus_t returnStatus = HTTP_SUCCESS;
+    char rangeValueBuffer[ MAX_RANGE_REQUEST_VALUE_LEN ] = { 0 };
+    size_t rangeValueLength = 0;
+    int stdRetVal = 0;
+
+    if( pRequestHeaders == NULL )
+    {
+        IotLogError( "Parameter check failed: pRequestHeaders is NULL." );
```

Comment:
Maybe change to "Parameter check failed: pRequestHeaders interface is NULL." for consistency.

Reply:
It should not say "interface"

Reply:
It is not an interface but an object.